python/metricq/connection_watchdog.py

```diff
@@ -74,19 +79,27 @@ def start(self):
 
             except CancelledError:
                 logger.debug("Cancelled {} watchdog", self.connection_name)
-                return
+                raise
```

Comment:
So, why do we need to propagate the `CancelledError`?

Reply:
As per [the documentation](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel):

> The coroutine then has a chance to clean up or even deny the request by suppressing the exception with a try … … except CancelledError … finally block. Therefore, unlike Future.cancel(), Task.cancel() does not guarantee that the Task will be cancelled, although **suppressing cancellation completely is not common and is actively discouraged**.

It can be argued, weather returning instead of reraising actually supresses cancellation.

I changed this initially to exclude it as a source of the issue I experienced. Turn out it is unrelated and it was probably fine all around. I just prefer to follow the common path if it doesn't hurt.

Reply:
Ohh, I see. Sure, reraising the exception is then probably the best idea. Thanks for the clarification.